src/Metadata/Resource/Factory/ResourceMetadataFactoryTrait.php

```diff
@@ -0,0 +1,125 @@
+<?php
+
+namespace ApiPlatform\Core\Metadata\Resource\Factory;
+
+use ApiPlatform\Core\Exception\ResourceClassNotFoundException;
+use ApiPlatform\Core\Metadata\Resource\Operation;
+use ApiPlatform\Core\Metadata\Resource\PaginationMetadata;
+use ApiPlatform\Core\Metadata\Resource\ResourceMetadata;
+use ApiPlatform\Core\Annotation\Resource;
+
+trait ResourceMetadataFactoryTrait
```

Comment:
This would indeed resolve the issue involving the use of `Annotation\Resource` with configuration values. 
Still, it's handy for the `handleNotFound` method.